tests/unit/Application/Assembler/ServiceStatusAssemblerTest.php

```diff
@@ -0,0 +1,265 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use Mockery as m;
+use Mockery\Adapter\Phpunit\MockeryTestCase;
+use Mockery\MockInterface;
+use Surfnet\ServiceProviderDashboard\Application\Assembler\ServiceStatusAssembler;
+use Surfnet\ServiceProviderDashboard\Application\Service\ServiceStatusService;
+use Surfnet\ServiceProviderDashboard\Domain\Entity\Service;
+use Surfnet\ServiceProviderDashboard\Domain\Entity\Entity;
+use Symfony\Component\Routing\RouterInterface;
+
+class ServiceStatusAssemblerTest extends MockeryTestCase
+{
+    /** @var Service|MockInterface */
+    private $service;
+
+    /** @var ServiceStatusService|MockInterface */
+    private $serviceStatusService;
+
+    /** @var RouterInterface|MockInterface */
+    private $router;
+
+    public function setUp()
+    {
+        $this->service = m::mock(Service::class);
+        $this->serviceStatusService = m::mock(ServiceStatusService::class);
+        $this->router = m::mock(RouterInterface::class);
+
+        $url = null;
+        $this->router
+            ->shouldReceive('generate')
+            ->andReturnUsing(function($route, $parameters) {
+                return $route.'?'.http_build_query($parameters);
+            });
+
+    }
+
+    /**
+     * @dataProvider assemblerProvider
+     */
+    public function test_it_can_return_the_status_of_a_service_($serviceData, $statusServiceData, $expectedJson)
+    {
+        $this->createServiceMock($serviceData);
+        $this->createStatusServiceMock($statusServiceData);
+
+        $assembler = new ServiceStatusAssembler(
+            $this->service,
+            $this->serviceStatusService,
+            $this->router
+        );
+
+        $result = $assembler->getDto();
+        $json = json_encode($result);
+
+        $this->assertSame($expectedJson, $json);
+    }
+
+    public static function assemblerProvider()
+    {
+        return [
+            'institute-output' => [
+                'service' =>[
+                    'id' => 2,
+                    'name' => 'service-2',
+                    'type' => Service::SERVICE_TYPE_INSTITUTE,
+                    'intakeStatus' => Service::INTAKE_STATUS_YES,
+                    'representativeApproved' => Service::SURFCONEXT_APPROVED_YES,
+                    'contractSigned' => null,
+                    'isPrivacyQuestionsEnabled' => true,
+                    'connectionStatus' => Service::CONNECTION_STATUS_ACTIVE,
+                    'entities' => [],
+                ],
+                'statusService' => [
+                    'entityStatus' => Service::ENTITY_PUBLISHED_YES,
```

Comment:
Maybe add an entity for good measure? Or set the entity status to no. Just for the sake of correctness.

Reply:
Added an entity